### ISSUE TITLE
🐛 [firebase_auth] multiFactor.getEnrolledFactors() throws an error on iOS

### ISSUE THREAD

**Issue by davidmigloz** [firebase/flutterfire#9363]
## Bug report

Given an iOS user with multi-factor enrolled, when the app calls `user.multiFactor.getEnrolledFactors()`, it throws `Null check operator used on a null value` error in the following line:

https://github.com/firebase/flutterfire/blob/2937fb8ec2abd5d823f5434150383e81fec80a05/packages/firebase_auth/firebase_auth_platform_interface/lib/src/pigeon/messages.pigeon.dart#L86

On Android it works as expected.

### Steps to reproduce

1. Enroll SMS multi-factor for a user
2. Call `user.multiFactor.getEnrolledFactors()`
3. Error is thrown

### Expected behavior

The method should return the list of enrolled factors.

---

## Additional context

Debugging the internals of the Flutter Firebase Auth SDK, I can see the following differences between iOS and Android in the `PhoneMultiFactorInfo` object for the same user:

Android:
<img width="378" alt="image" src="https://user-images.githubusercontent.com/6546265/184918650-3f4e73dd-5dbf-435a-8d3c-af2b3078897a.png">

iOS:
<img width="416" alt="image" src="https://user-images.githubusercontent.com/6546265/184918732-a37c3279-cb77-45c1-b4fd-e517c3522646.png">

- `enrollmentTimestamp` is 0 in Android (known bug: https://github.com/firebase/flutterfire/issues/9354)
- `factorID` is nil in iOS

As `factorID` is nil, when pigeon tries to deserialize it, it throws the "Null check operator used on a null value" error.
```
factorId: pigeonMap['factorId']! as String,
```

---

### Flutter doctor

Run `flutter doctor` and paste the output below:

<details><summary>Click To Expand</summary>

```
❯ flutter doctor
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.0.5, on macOS 12.5 21G72 darwin-arm, locale
    en-NL)
[✓] Android toolchain - develop for Android devices (Android SDK version
    32.1.0-rc1)
[✓] Xcode - develop for iOS and macOS (Xcode 13.4.1)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2021.2)
[✓] Android Studio (version 2021.2)
[✓] IntelliJ IDEA Ultimate Edition (version 2022.1.2)
[✓] IntelliJ IDEA Ultimate Edition (version 2022.1.2)
[✓] VS Code (version 1.70.0)
[✓] Connected device (4 available)
[✓] HTTP Host Availability

• No issues found!
```

</details>

---

### Flutter dependencies

Run `flutter pub deps -- --style=compact` and paste the output below:

<details><summary>Click To Expand</summary>

```
Dart SDK 2.17.6
Flutter SDK 3.0.5
brxs_app 0.0.1+1

dependencies:
- app_passcode 0.0.1 [app_lock common_dependencies_feature shared_profile]
- common_dependencies_feature 0.0.1 [flutter app_build_config app_lock authentication brxs_design_system countries dart_toolkit data_layer_common flutter_toolkit injection localization money navigation resolver cached_network_image dio equatable intl flutter_bloc flutter_keyboard_visibility flutter_svg rxdart url_launcher]
- deal 0.0.1 [common_dependencies_feature shared_profile shared_properties shared_support fl_chart]
- debug_config 0.0.1 [common_dependencies_feature shared_profile]
- device_preview 1.1.0 [flutter flutter_localizations provider device_frame freezed_annotation json_annotation shared_preferences collection]
- firebase_app 0.0.1 [flutter app_build_config firebase_core]
- firebase_messaging 12.0.2 [firebase_core firebase_core_platform_interface firebase_messaging_platform_interface firebase_messaging_web flutter meta]
- flutter 0.0.0 [characters collection material_color_utilities meta vector_math sky_engine]
- home 0.0.1 [common_dependencies_feature shared_support]
- log_in 0.0.1 [common_dependencies_feature shared_profile]
- marketplace 0.0.1 [common_dependencies_feature shared_properties]
- multi_factor_auth 0.0.1 [common_dependencies_feature shared_profile]
- portfolio 0.0.1 [common_dependencies_feature]
- referral 0.0.1 [common_dependencies_feature]
- setup_profile 0.0.1 [common_dependencies_feature shared_profile]
- shared_active_scaffold 0.0.1 [common_dependencies_feature home portfolio marketplace shared_profile support]
- shared_preferences 2.0.15 [flutter shared_preferences_android shared_preferences_ios shared_preferences_linux shared_preferences_macos shared_preferences_platform_interface shared_preferences_web shared_preferences_windows]
- shared_profile 0.0.1 [api_profile_v1 app_lock common_dependencies_feature]
- shared_properties 0.0.1 [api_property_v1 api_performance_v1 common_dependencies_feature]
- shared_support 0.0.1 [api_content_v1 common_dependencies_feature]
- sign_up 0.0.1 [common_dependencies_feature launch_mail_app shared_profile]
- splash 0.0.1 [common_dependencies_feature]
- support 0.0.1 [common_dependencies_feature shared_support]
- theme_manager 0.0.1 [flutter brxs_design_system flutter_bloc shared_preferences]
- welcome 0.0.1 [common_dependencies_feature debug_config]

dev dependencies:
- flutter_lints 2.0.1 [lints]
- flutter_test 0.0.0 [flutter test_api path fake_async clock stack_trace vector_math async boolean_selector characters charcode collection matcher material_color_utilities meta source_span stream_channel string_scanner term_glyph]
- integration_test 0.0.0 [flutter flutter_driver flutter_test path vm_service archive async boolean_selector characters charcode clock collection crypto fake_async file matcher material_color_utilities meta source_span stack_trace stream_channel string_scanner sync_http term_glyph test_api typed_data vector_math webdriver]

dependency overrides:
- api_content_v1 1.0.0 [dio json_annotation]
- api_performance_v1 1.0.0 [dio json_annotation]
- api_profile_v1 1.0.0 [dio json_annotation]
- api_property_v1 1.0.0 [dio json_annotation]
- app_build_config 0.0.1 [flutter_bloc meta shared_preferences]
- app_lock 0.0.1 [dart_toolkit flutter flutter_web_plugins plugin_platform_interface rxdart shared_preferences url_launcher]
- app_passcode 0.0.1 [app_lock common_dependencies_feature shared_profile]
- authentication 0.0.1 [flutter dart_toolkit firebase_auth equatable meta rxdart shared_preferences]
- brxs_design_system 0.0.1 [flutter countries dart_toolkit flutter_toolkit localization money theme_manager equatable flutter_markdown flutter_svg go_router]
- common_dependencies_feature 0.0.1 [flutter app_build_config app_lock authentication brxs_design_system countries dart_toolkit data_layer_common flutter_toolkit injection localization money navigation resolver cached_network_image dio equatable intl flutter_bloc flutter_keyboard_visibility flutter_svg rxdart url_launcher]
- common_dev_dependencies_feature 0.0.1 [bloc_test flutter_test integration_test mocktail test]
- countries 0.0.1 [flutter flutter_localizations meta]
- dart_toolkit 0.0.1 [built_collection collection fuzzy intl meta]
- data_layer_common 0.0.1 [flutter authentication countries dart_toolkit money dio meta rxdart]
- deal 0.0.1 [common_dependencies_feature shared_profile shared_properties shared_support fl_chart]
- debug_config 0.0.1 [common_dependencies_feature shared_profile]
- firebase_app 0.0.1 [flutter app_build_config firebase_core]
- flutter_toolkit 0.0.1 [flutter flutter_custom_tabs meta sliver_tools url_launcher]
- home 0.0.1 [common_dependencies_feature shared_support]
- injection 0.0.1 [app_build_config get_it]
- launch_mail_app 0.0.1 [flutter flutter_web_plugins plugin_platform_interface url_launcher]
- localization 0.0.1 [flutter flutter_localizations dart_toolkit intl flutter_bloc shared_preferences]
- log_in 0.0.1 [common_dependencies_feature shared_profile]
- marketplace 0.0.1 [common_dependencies_feature shared_properties]
- money 0.0.1 [meta intl]
- multi_factor_auth 0.0.1 [common_dependencies_feature shared_profile]
- navigation 0.0.1 [flutter go_router]
- portfolio 0.0.1 [common_dependencies_feature]
- referral 0.0.1 [common_dependencies_feature]
- resolver 0.0.1 [injection]
- setup_profile 0.0.1 [common_dependencies_feature shared_profile]
- shared_active_scaffold 0.0.1 [common_dependencies_feature home portfolio marketplace shared_profile support]
- shared_profile 0.0.1 [api_profile_v1 app_lock common_dependencies_feature]
- shared_properties 0.0.1 [api_property_v1 api_performance_v1 common_dependencies_feature]
- shared_support 0.0.1 [api_content_v1 common_dependencies_feature]
- sign_up 0.0.1 [common_dependencies_feature launch_mail_app shared_profile]
- splash 0.0.1 [common_dependencies_feature]
- support 0.0.1 [common_dependencies_feature shared_support]
- theme_manager 0.0.1 [flutter brxs_design_system flutter_bloc shared_preferences]
- welcome 0.0.1 [common_dependencies_feature debug_config]

transitive dependencies:
- _fe_analyzer_shared 40.0.0 [meta]
- analyzer 4.1.0 [_fe_analyzer_shared collection convert crypto glob meta package_config path pub_semver source_span watcher yaml]
- archive 3.1.11 [crypto path]
- args 2.3.1
- async 2.8.2 [collection meta]
- bloc 8.0.3 [meta]
- bloc_test 9.0.3 [bloc diff_match_patch meta mocktail test]
- boolean_selector 2.1.0 [source_span string_scanner]
- built_collection 5.1.1
- cached_network_image 3.2.1 [flutter flutter_cache_manager octo_image cached_network_image_platform_interface cached_network_image_web]
- cached_network_image_platform_interface 1.0.0 [flutter flutter_cache_manager]
- cached_network_image_web 1.0.1 [flutter flutter_cache_manager cached_network_image_platform_interface]
- characters 1.2.0
- charcode 1.3.1
- clock 1.1.0
- collection 1.16.0
- convert 3.0.1 [typed_data]
- coverage 1.2.0 [args logging package_config path source_maps stack_trace vm_service]
- crypto 3.0.1 [collection typed_data]
- device_frame 1.1.0 [freezed_annotation flutter]
- diff_match_patch 0.4.1
- dio 4.0.6 [http_parser path]
- equatable 2.0.3 [collection meta]
- fake_async 1.3.0 [clock collection]
- ffi 1.1.2
- file 6.1.2 [meta path]
- firebase_auth 3.6.3 [firebase_auth_platform_interface firebase_auth_web firebase_core firebase_core_platform_interface flutter meta]
- firebase_auth_platform_interface 6.5.3 [collection firebase_core flutter meta plugin_platform_interface]
- firebase_auth_web 4.2.3 [firebase_auth_platform_interface firebase_core firebase_core_web flutter flutter_web_plugins http_parser intl js meta]
- firebase_core 1.20.1 [firebase_core_platform_interface firebase_core_web flutter meta]
- firebase_core_platform_interface 4.5.0 [collection flutter flutter_test meta plugin_platform_interface]
- firebase_core_web 1.7.1 [firebase_core_platform_interface flutter flutter_web_plugins js meta]
- firebase_messaging_platform_interface 4.1.1 [firebase_core flutter meta plugin_platform_interface]
- firebase_messaging_web 3.1.1 [firebase_core firebase_core_web firebase_messaging_platform_interface flutter flutter_web_plugins js meta]
- fl_chart 0.55.0 [flutter equatable]
- flutter_bloc 8.0.1 [flutter bloc provider]
- flutter_blurhash 0.7.0 [flutter]
- flutter_cache_manager 3.3.0 [clock collection file flutter http path path_provider pedantic rxdart sqflite uuid]
- flutter_custom_tabs 1.0.4 [flutter flutter_custom_tabs_platform_interface flutter_custom_tabs_web meta]
- flutter_custom_tabs_platform_interface 1.0.1 [flutter plugin_platform_interface meta]
- flutter_custom_tabs_web 1.0.0 [flutter flutter_web_plugins flutter_custom_tabs_platform_interface meta url_launcher_web url_launcher_platform_interface]
- flutter_driver 0.0.0 [file flutter flutter_test fuchsia_remote_debug_protocol path meta vm_service webdriver archive async boolean_selector characters charcode clock collection crypto matcher material_color_utilities platform process source_span stack_trace stream_channel string_scanner sync_http term_glyph test_api typed_data vector_math]
- flutter_keyboard_visibility 5.3.0 [meta flutter_keyboard_visibility_platform_interface flutter_keyboard_visibility_web flutter]
- flutter_keyboard_visibility_platform_interface 2.0.0 [flutter meta plugin_platform_interface]
- flutter_keyboard_visibility_web 2.0.0 [flutter_keyboard_visibility_platform_interface flutter_web_plugins flutter]
- flutter_localizations 0.0.0 [flutter intl characters clock collection material_color_utilities meta path vector_math]
- flutter_markdown 0.6.10+3 [flutter markdown meta path]
- flutter_svg 1.1.1+1 [flutter meta path_drawing vector_math xml]
- flutter_web_plugins 0.0.0 [flutter js characters collection material_color_utilities meta vector_math]
- freezed_annotation 2.0.3 [collection json_annotation meta]
- frontend_server_client 2.1.3 [async path]
- fuchsia_remote_debug_protocol 0.0.0 [process vm_service file meta path platform]
- fuzzy 0.4.0-nullsafety.0 [latinize]
- get_it 7.2.0 [async collection]
- glob 2.0.2 [async collection file path string_scanner]
- go_router 4.2.4 [collection flutter flutter_web_plugins logging meta]
- http 0.13.4 [async http_parser meta path]
- http_multi_server 3.2.0 [async]
- http_parser 4.0.0 [charcode collection source_span string_scanner typed_data]
- intl 0.17.0 [clock path]
- io 1.0.3 [meta path string_scanner]
- js 0.6.4
- json_annotation 4.5.0 [meta]
- latinize 0.1.0-nullsafety.0
- lints 2.0.0
- logging 1.0.2
- markdown 5.0.0 [args charcode meta]
- matcher 0.12.11 [stack_trace]
- material_color_utilities 0.1.4
- meta 1.7.0
- mime 1.0.2
- mocktail 0.3.0 [collection matcher test]
- nested 1.0.0 [flutter]
- node_preamble 2.0.1
- octo_image 1.0.2 [flutter flutter_blurhash]
- package_config 2.0.2 [path]
- path 1.8.1
- path_drawing 1.0.0 [vector_math meta path_parsing flutter]
- path_parsing 1.0.0 [vector_math meta]
- path_provider 2.0.11 [flutter path_provider_android path_provider_ios path_provider_linux path_provider_macos path_provider_platform_interface path_provider_windows]
- path_provider_android 2.0.14 [flutter path_provider_platform_interface]
- path_provider_ios 2.0.10 [flutter path_provider_platform_interface]
- path_provider_linux 2.1.5 [ffi flutter path path_provider_platform_interface xdg_directories]
- path_provider_macos 2.0.6 [flutter path_provider_platform_interface]
- path_provider_platform_interface 2.0.3 [flutter platform plugin_platform_interface]
- path_provider_windows 2.0.5 [ffi flutter path path_provider_platform_interface win32]
- pedantic 1.11.1
- petitparser 5.0.0 [meta]
- platform 3.1.0
- plugin_platform_interface 2.1.2 [meta]
- pool 1.5.0 [async stack_trace]
- process 4.2.4 [file path platform]
- provider 6.0.2 [collection flutter nested]
- pub_semver 2.1.1 [collection meta]
- rxdart 0.27.5
- shared_preferences_android 2.0.11 [flutter shared_preferences_platform_interface]
- shared_preferences_ios 2.1.0 [flutter shared_preferences_platform_interface]
- shared_preferences_linux 2.1.0 [file flutter path path_provider_linux path_provider_platform_interface shared_preferences_platform_interface]
- shared_preferences_macos 2.0.3 [flutter shared_preferences_platform_interface]
- shared_preferences_platform_interface 2.0.0 [flutter]
- shared_preferences_web 2.0.3 [flutter flutter_web_plugins shared_preferences_platform_interface]
- shared_preferences_windows 2.1.0 [file flutter path path_provider_platform_interface path_provider_windows shared_preferences_platform_interface]
- shelf 1.3.0 [async collection http_parser path stack_trace stream_channel]
- shelf_packages_handler 3.0.0 [path shelf shelf_static]
- shelf_static 1.1.0 [convert http_parser mime path shelf]
- shelf_web_socket 1.0.1 [shelf stream_channel web_socket_channel]
- sky_engine 0.0.99
- sliver_tools 0.2.7 [flutter]
- source_map_stack_trace 2.1.0 [path stack_trace source_maps]
- source_maps 0.10.10 [source_span]
- source_span 1.8.2 [collection path term_glyph]
- sqflite 2.0.2+1 [flutter sqflite_common path]
- sqflite_common 2.2.1+1 [synchronized path meta]
- stack_trace 1.10.0 [path]
- stream_channel 2.1.0 [async]
- string_scanner 1.1.0 [charcode source_span]
- sync_http 0.3.0
- synchronized 3.0.0+2
- term_glyph 1.2.0
- test 1.21.1 [analyzer async boolean_selector collection coverage http_multi_server io js node_preamble package_config path pool shelf shelf_packages_handler shelf_static shelf_web_socket source_span stack_trace stream_channel typed_data web_socket_channel webkit_inspection_protocol yaml test_api test_core]
- test_api 0.4.9 [async boolean_selector collection meta source_span stack_trace stream_channel string_scanner term_glyph matcher]
- test_core 0.4.13 [analyzer async args boolean_selector collection coverage frontend_server_client glob io meta package_config path pool source_map_stack_trace source_maps source_span stack_trace stream_channel vm_service yaml matcher test_api]
- typed_data 1.3.0 [collection]
- url_launcher 6.1.5 [flutter url_launcher_android url_launcher_ios url_launcher_linux url_launcher_macos url_launcher_platform_interface url_launcher_web url_launcher_windows]
- url_launcher_android 6.0.17 [flutter url_launcher_platform_interface]
- url_launcher_ios 6.0.17 [flutter url_launcher_platform_interface]
- url_launcher_linux 3.0.1 [flutter url_launcher_platform_interface]
- url_launcher_macos 3.0.1 [flutter url_launcher_platform_interface]
- url_launcher_platform_interface 2.1.0 [flutter plugin_platform_interface]
- url_launcher_web 2.0.11 [flutter flutter_web_plugins url_launcher_platform_interface]
- url_launcher_windows 3.0.1 [flutter url_launcher_platform_interface]
- uuid 3.0.6 [crypto]
- vector_math 2.1.2
- vm_service 8.2.2
- watcher 1.0.1 [async path]
- web_socket_channel 2.2.0 [async crypto stream_channel]
- webdriver 3.0.0 [archive matcher path stack_trace sync_http]
- webkit_inspection_protocol 1.1.0 [logging]
- win32 2.5.2 [ffi]
- xdg_directories 0.2.0+1 [meta path process]
- xml 6.1.0 [collection meta petitparser]
- yaml 3.1.1 [collection source_span string_scanner]
```

</details>

---

**Comment by Lyokone** [comment]
Hello,
Thanks for the bug report, I'll have a look now

**Comment by davidmigloz** [comment]
Thanks for the quick fix!

**Comment by Lyokone** [comment]
Fixed in `firebase_auth: ^3.6.4`
